src/components/show-hide-when/display-when.ts

```diff
@@ -1,46 +1,134 @@
+      this.checkConditions(query);
+    }
+  }
+
+  get when(): string {
+    return this._query;
+  }
+
+  constructor(conditions: string, public _config: Config, public _plt: Platform, public zone: NgZone) {
+    // this.checkConditions(conditions);
+  }
+
+  checkConditions(conditions: string): void {
     if (!conditions) return;
 
-    this.conditions = conditions.replace(/\s/g, '').split(',');
+    // Split the conditions on commas, unless the comma is inside of parantheses
+    // because this means it is inside of a media query
+    this.conditions = conditions.replace(/\s/g, '').split(/\,(?![^\(]*\))/g);
+    console.log('conditions', this.conditions);
 
     // check if its one of the matching platforms first
     // a platform does not change during the life of an app
     for (let i = 0; i < this.conditions.length; i++) {
-      if (this.conditions[i] && _plt.is(this.conditions[i])) {
-        this.isMatch = true;
-        return;
+      console.log('condition', this.conditions[i]);
+
+      // media: xs, media: (min-width: 44px), media: print and (min-width: 450px)
+      if (this.conditions[i] && this.conditions[i].indexOf('media') > -1) {
+        this.compareMedia(this.conditions[i]);
+      }
+
+      // mode: ios, mode: wp, mode: md
+      if (this.conditions[i] && this.conditions[i].indexOf('mode') > -1) {
+        this.compareMode(this.conditions[i]);
+      }
+
+      // platform: ios, platform: android, platform: iphone
+      // The expression after the OR statement is for backwards-compatibility
+      if (this.conditions[i] && (this.conditions[i].indexOf('platform') > -1) || (this._plt.is(this.conditions[i]))) {
+        this.comparePlatform(this.conditions[i]);
       }
     }
 
+    // orientation: portrait, orientation: landscape
     if (this.orientation()) {
       // add window resize listener
-      this.resizeObs = _plt.resize.subscribe(this.orientation.bind(this));
+      this._resizeObs = this._plt.resize.subscribe(this.orientation.bind(this));
     }
+  }
 
+  compareMedia(condition: string): void {
+    condition = condition.replace('media:', '');
+
+    let query = QUERY[condition];
+    query = (query) ? query : condition;
+
+    if (query.indexOf(',') > -1 || query.indexOf('and') > -1) {
+      console.log('remove outside parens');
+      // query = query.substring(1, query.length - 1);
+    }
+
+    if (query && query.length > 0) {
+      // Listen
+      const callback = (query: MediaQueryList) => this.isMatch = query.matches;
+      const mediaList = this._plt.win().matchMedia(query);
+
+      console.log('mediaList', mediaList);
+
+      mediaList.addListener(callback);
+      this.isMatch = mediaList.matches;
+      this._rmListener = function () {
+        mediaList.removeListener(callback);
+      };
+    }
+  }
+
+  compareMode(condition: string): void {
+    condition = condition.replace('mode:', '');
+    if (condition === this._config.get('mode')) {
+      this.isMatch = true;
```

Comment:
can we add a callback so it is possible to hook any change to `this.isMatch`? right now we are relying on change detection, but it would be interesting to implement some reactive approach instead.